src/valhalla_service.cc

```diff
@@ -19,48 +21,82 @@ using namespace prime_server;
 #include "tyr/actor.h"
 
 int main(int argc, char** argv) {
+  const auto program = filesystem::path(__FILE__).stem().string();
+  std::vector<std::string> pos_args;
+  boost::property_tree::ptree config;
+
+  cxxopts::Options options(
+      program,
+      program + " " + VALHALLA_PRINT_VERSION +
+          "\n\n"
+          "a program to run a multi-threaded Valhalla HTTP service powered by https://github.com/kevinkreiser/prime_server\n");
+
+  // clang-format off
+  options.add_options()
+    ("h,help", "Print this help message.")
+    ("v,version","Print the version of this software.")
+    // no optional args bcs of pre-cxxopts backwards-compatibility
+    ("pos_args", "positional arguments", cxxopts::value<std::vector<std::string>>(pos_args));
+  // clang-format on
+
+  try {
+    options.parse_positional({"pos_args"});
+    options.positional_help("CONFIG_JSON [CONCURRENCY] or CONFIG_JSON ACTION JSON_REQUEST");
+    auto result = options.parse(argc, argv);
+    // We set up conf & num_threads ourselves
+    if (!parse_common_args(program, options, result, nullptr, "", false))
+      return EXIT_SUCCESS;
+
 #ifdef ENABLE_SERVICES
-  if (argc < 2 || argc > 4) {
-    LOG_ERROR("Usage: " + std::string(argv[0]) + " config/file.json [concurrency]");
-    LOG_ERROR("Usage: " + std::string(argv[0]) + " config/file.json action json_request");
-    return 1;
-  }
+    if (pos_args.size() < 1 || pos_args.size() > 3 || pos_args.size() == 2) {
+      throw cxxopts::exceptions::exception("[FATAL] Too many or few arguments, see --help:\n");
+    }
```

Comment:
`pos_args.size() == 2` breaks the ability to set a concurrency like this:

```
./valhalla_service ./valhalla_data/config.json 1
[FATAL] Too many or few arguments, see --help:

valhalla_service 3.5.1

a program to run a multi-threaded Valhalla HTTP service powered by https://github.com/kevinkreiser/prime_server

Usage:
  valhalla_service [OPTION...] CONFIG_JSON [CONCURRENCY] or CONFIG_JSON ACTION JSON_REQUEST

  -h, --help     Print this help message.
  -v, --version  Print the version of this software.
```

Reply:
ah yeah we have to fix this, good catch